test/metabase/api/table_test.clj

```diff
@@ -493,42 +491,40 @@
-                                            :base_type    "type/Text"
+                                          [{:name           "NAME"
+                                            :display_name   "NAME"
+                                            :base_type      "type/Text"
                                             :effective_type "type/Text"
-                                            :semantic_type "type/Name"
-                                            :fingerprint  (:name mock.util/venue-fingerprints)
-                                            :field_ref    ["field" "NAME" {:base-type "type/Text"}]}
-                                           {:name         "ID"
-                                            :display_name "ID"
-                                            :base_type    "type/BigInteger"
+                                            :semantic_type  "type/Name"
+                                            :fingerprint    (:fingerprint (mt/col :venues :name))
+                                            :field_ref      ["field" "NAME" {:base-type "type/Text"}]}
+                                           {:name           "ID"
+                                            :display_name   "ID"
+                                            :base_type      "type/BigInteger"
                                             :effective_type "type/BigInteger"
-                                            :semantic_type nil
-                                            :fingerprint  (:id mock.util/venue-fingerprints)
-                                            :field_ref    ["field" "ID" {:base-type "type/BigInteger"}]}
+                                            :semantic_type  nil
+                                            :fingerprint    (:fingerprint (mt/col :venues :id))
+                                            :field_ref      ["field" "ID" {:base-type "type/BigInteger"}]}
                                            (with-numeric-dimension-options
-                                             {:name         "PRICE"
-                                              :display_name "PRICE"
-                                              :base_type    "type/Integer"
+                                             {:name           "PRICE"
+                                              :display_name   "PRICE"
+                                              :base_type      "type/Integer"
                                               :effective_type "type/Integer"
-                                              :semantic_type nil
-                                              :fingerprint  (:price mock.util/venue-fingerprints)
-                                              :field_ref    ["field" "PRICE" {:base-type "type/Integer"}]})
+                                              :semantic_type  nil
+                                              :fingerprint    (:fingerprint (mt/col :venues :price))
```

Comment:
Looks like this repeated call to `mt/col` does lots of `(db/select-one [Field :id :table_id :semantic_type :base_type :effective_type ...])`. Can we get the fields for a table and then access that multiple times? Or maybe a `TODO:` for the next PR so this doesn't have lots of non-mechanical changes?

Reply:
Yeah that's a good point, I should probably just use `db/id->field` to get a map of Field ID -> `:fingerprint` and do things from there. That's what I did in a few other places

Reply:
fixed